stan/math/rev/core/var.hpp

```diff
@@ -285,7 +188,7 @@ class var {
    *
    * @return variable
    */
-  inline vari& operator*() { return *vi_; }
+  inline vari_type& operator*() { return *vi_; }
```

Comment:
This is weird operator. I wonder what these are used for? I guess it's outside the scope of this pull, but I wouldn't mind if we removed it if it's not used.

Reply:
Yeah it's a bit odd, but tbh idt it hurts anything so it's probs fine to leave